### ISSUE TITLE
Provide expression function metadata via REST

### ISSUE THREAD

**Issue by igarashitm** [atlasmap/atlasmap#2093]
`AtlasService` should provide function metadata via REST service so that UI can use those info for expression syntax assist, like description, parameter type and return type shown as a tooltip on a function, etc. We'd add an annotation for that, just like `AtlasActionProcessor` for transformation.

**Comment by igarashitm** [comment]
@pleacu I almost forgot the #3138 - so in order to implement language server for the conditional mapping expression, function metadata would also have to align with the protocol. The keywords and operators would also eventually migrate to server side on that route. Did you get anything around that so far?

**Comment by pleacu** [comment]
I was expecting that there would be a REST API for attaining the operators, keywords and field action functions.  The meta-data for the functions would affect the presentation of those elements within the UI.  I'm not sure if I know anywhere in the UI where there's actual reliance on parsing logic.  I guess we can find that along the way.  Is there a formal LSP that you're thinking of?

**Comment by igarashitm** [comment]
Do you mean this? https://microsoft.github.io/language-server-protocol/
This is what I only got ATM, but I expect by following LSP, we can leverage more features of monaco-editor, like syntax highlight, function completion hovering help for the functions, etc. And to do that, only returning AtlasMap specific function metadata won't work, but need to follow the protocol I guess.

**Comment by igarashitm** [comment]
Putting on hold - research on #3138 has to be done before this.

**Comment by pleacu** [comment]
I'd love to see an example of the Microsoft LSP just to see if it's worth the overhead/ complexity.

**Comment by igarashitm** [comment]
will be tracked on #3138
